src/main/java/com/google/cloud/tools/gradle/endpoints/client/EndpointsClientPlugin.java

```diff
@@ -0,0 +1,183 @@
+            // make sure we depend on the server configuration build tasks, so those get run
+            // before we run our task to get the discovery docs
+            extractDiscoveryDocs.dependsOn(
+                project.getConfigurations().getByName(ENDPOINTS_SERVER_CONFIGURATION)
+                    .getBuildDependencies());
+
+            // iterate through the configuration and get all artifacts (discovery doc zips)
+            project.afterEvaluate(new Action<Project>() {
+              @Override
+              public void execute(Project project) {
+                Collection<File> files = project.getConfigurations()
+                    .getByName(ENDPOINTS_SERVER_CONFIGURATION).getFiles();
+                extractDiscoveryDocs.setDiscoveryDocZips(files);
+                extractDiscoveryDocs.setDiscoveryDocsDir(extension.getGenDiscoveryDocsDir());
+              }
+            });
+          }
+        });
+  }
+
+  private void createGenerateClientLibTask() {
+    project.getTasks().create(GENERATE_CLIENT_LIBRARY_TASK, GenerateClientLibrariesTask.class,
+        new Action<GenerateClientLibrariesTask>() {
+          @Override
+          public void execute(final GenerateClientLibrariesTask genClientLibs) {
+            genClientLibs.setDescription("_internal");
+            genClientLibs.dependsOn(EXTRACT_SERVER_DISCOVERY_DOCS_TASK);
+
+            project.afterEvaluate(new Action<Project>() {
+              @Override
+              public void execute(Project project) {
+                genClientLibs.setClientLibraryDir(extension.getClientLibDir());
+                genClientLibs.setDiscoveryDocs(getDiscoveryDocs());
+                genClientLibs.setGeneratedDiscoveryDocs(extension.getGenDiscoveryDocsDir());
+              }
+            });
+          }
+        });
+  }
+
+  // find discovery docs from files and directories
+  private List<File> getDiscoveryDocs() {
+    List<File> discoveryDocs = new ArrayList<File>();
+    for (File discoveryDoc : extension.getDiscoveryDocs()) {
+      if (discoveryDoc.isDirectory()) {
+        discoveryDocs.addAll(DiscoveryDocUtil.findDiscoveryDocsInDirectory(discoveryDoc));
+      }
+      else {
+        discoveryDocs.add(discoveryDoc);
+      }
+    }
+    return discoveryDocs;
+  }
+
+  private void createGenerateClientLibSrcTask() {
+    project.getTasks()
+        .create(GENERATE_CLIENT_LIBRARY_SRC_TASK, GenerateClientLibrarySourceTask.class,
+            new Action<GenerateClientLibrarySourceTask>() {
+              @Override
+              public void execute(final GenerateClientLibrarySourceTask genClientLibSrc) {
+                genClientLibSrc.setDescription("_internal");
+                genClientLibSrc.dependsOn(GENERATE_CLIENT_LIBRARY_TASK);
+
+                project.afterEvaluate(new Action<Project>() {
+                  @Override
+                  public void execute(Project project) {
+                    genClientLibSrc.setClientLibDir(extension.getClientLibDir());
```

Comment:
one is the where the zips go, and one is a src directory that is registered with gradle